docs/internals/layout_in_storage.rst

```diff
@@ -127,16 +127,20 @@ The type of the value is ``uint256``, so it uses a single slot.
 ``bytes`` and ``string``
 ------------------------
 
-``bytes`` and ``string`` are encoded identically.
-In general, the encoding is similar to ``bytes1[]``, in the sense that there is a slot for the array itself and
-a data area that is computed using a ``keccak256`` hash of that slot's position.
-However, for short values (shorter than 32 bytes) the array elements are stored together with the length in the same slot.
+``bytes`` and ``string`` are encoded identically. In general, the encoding is similar to ``bytes1[]`` in the
+ sense that the layout in storage depends on the ``.length``:
```

Comment:
But this is not true. The layout of `bytes1[]` does not depend on `.length`. The similarity is that the long form of `bytes` and `bytes1[]` both have a length slot and a separate data area.